sway/tree/view.c

```diff
@@ -785,9 +785,21 @@ void view_map(struct sway_view *view, struct wlr_surface *wlr_surface,
 
 	view_execute_criteria(view);
 
+
+#if HAVE_XWAYLAND
+	if (wlr_surface_is_xwayland_surface(wlr_surface)) {
+		if (wlr_xwayland_or_surface_wants_focus(
```

Comment:
`wlr_xwayland_or_surface_wants_focus` is for override-redirect windows only (that's why there's "or" in the name), so I'm not comfortable using it for regular windows.

Maybe we could add a wlroots helper that returns the input model for a surface (as described in https://tronche.com/gui/x/icccm/sec-4.html#s-4.1.7)?

Reply:
Hey no worries.  I missed that comment.  I read it simply as "or" :smile: